### ISSUE TITLE
terraform_docs sed issue with v1.93.1 for mac default sed

### ISSUE THREAD

**Issue by BeneHa** [antonbabenko/pre-commit-terraform#707]
### Describe the bug

Since the fix of a bug in https://github.com/antonbabenko/pre-commit-terraform/pull/704 by @cschroer , the empty sed -i flag does not work for me. Now, a `README.md-e` file is being created as well as the `README.md`. I am using Mac's default sed.

I know I can install GNU sed, but I think many users are using the default sed and will run into this issue.

### How can we reproduce it?

Run pre-commit tf_docs on a Mac with the default sed installed.

### Environment information

* OS:  
```bash
Darwin 2BY1-1GLP3 23.6.0 Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000 arm64
```
* sed:
```bash
sed_version="$(sw_vers -productVersion)"
echo $sed_version
14.6.1
```

* Tools availability and versions:

```bash
GNU bash, version 5.2.32(1)-release (aarch64-apple-darwin23.4.0)
pre-commit 3.8.0
zsh: command not found: tofu
Terraform v1.9.5
python SKIPPED
Python 3.12.4
checkov SKIPPED
infracost SKIPPED
terraform-docs version v0.18.0 228c7a7 darwin/arm64
terragrunt SKIPPED
terrascan SKIPPED
TFLint version 0.53.0
+ ruleset.terraform (0.9.1-bundled)
tfsec SKIPPED
trivy SKIPPED
tfupdate SKIPPED
hcledit SKIPPED
```

* `.pre-commit-config.yaml`:

<details><summary>file content</summary>

```yaml
repos:
  - repo: https://github.com/antonbabenko/pre-commit-terraform.git
    rev: v1.94.0
    hooks:
      - id: terraform_docs
        args:
          - --hook-config=--create-file-if-not-exist=true
```
</details>

**Comment by robinbowes** [comment]
I can confirm this behaviour:

```
❯ ls
tt.txt

❯ gsed -i'' -e 's/foo/foobar/' tt.txt

❯ ls
tt.txt

❯ sed -i'' -e 's/foo/foobar/' tt.txt

❯ ls
tt.txt   tt.txt-e
```

Some useful information here about replacing strings in files: http://mywiki.wooledge.org/BashFAQ/021

tl;dr: "The preferred way to modify a file is to create a new file within the same file system, write the modified content into it, and then mv it to the original name."

**Comment by MaxymVlasov** [comment]
I think Apply should offer $9.99/month subscription for just replacing OSX with something like ElementaryOS with some integrations top :D 

If serious, please let me know is next work on Mac or not

```
repos:
  - repo: https://github.com/antonbabenko/pre-commit-terraform
    rev: aee5dc9a8d6625a47396f2b4797368c57cbb5f49
    hooks:
      - id: terraform_docs
```

**Comment by MaxymVlasov** [comment]
Test this please
```yaml
repos:
  - repo: https://github.com/antonbabenko/pre-commit-terraform
    rev: a76f03bfc4978741f37f79a6b4a86dc72078a78d
    hooks:
      - id: terraform_docs
```

**Comment by drAlberT** [comment]
> Test this please
> 
> ```yaml
> repos:
>   - repo: https://github.com/antonbabenko/pre-commit-terraform
>     rev: a76f03bfc4978741f37f79a6b4a86dc72078a78d
>     hooks:
>       - id: terraform_docs
> ```

Yes it does

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.94.1 :tada:
